client/jetpack-cloud/sections/partner-portal/primary/payment-method-add/index.tsx

```diff
@@ -1,13 +1,106 @@
-import { Card } from '@automattic/components';
+/**
+ * External dependencies
+ */
+import React, { ReactElement, useState, useMemo, useEffect, useCallback } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import page from 'page';
+import { StripeHookProvider, useStripe } from '@automattic/calypso-stripe';
 import { useTranslate } from 'i18n-calypso';
-import { ReactElement } from 'react';
+import { CheckoutProvider, CheckoutSubmitButton } from '@automattic/composite-checkout';
+import { Button, Card } from '@automattic/components';
+
+/**
+ * Internal dependencies
+ */
+import Main from 'calypso/components/main';
 import CardHeading from 'calypso/components/card-heading';
 import DocumentHead from 'calypso/components/data/document-head';
-import Main from 'calypso/components/main';
 import SidebarNavigation from 'calypso/jetpack-cloud/sections/partner-portal/sidebar-navigation';
+import { useCreateStoredCreditCardMethod } from 'calypso/jetpack-cloud/sections/partner-portal/payment-methods/hooks/use-create-stored-credit-card';
+import { getCurrentUserLocale } from 'calypso/state/current-user/selectors';
+import { getAllStoredCards } from 'calypso/state/stored-cards/selectors';
+import { getStripeConfiguration } from 'calypso/lib/store-transactions';
+import { errorNotice, infoNotice, successNotice } from 'calypso/state/notices/actions';
+import {
+	assignNewCardProcessor,
+	NewCardSubmitData,
+} from 'calypso/jetpack-cloud/sections/partner-portal/payment-methods/assignment-processor-functions';
+import type { TranslateResult } from 'i18n-calypso';
+import FormInputCheckbox from 'calypso/components/forms/form-checkbox';
+import FormLabel from 'calypso/components/forms/form-label';
+import CreditCardLoading from 'calypso/jetpack-cloud/sections/partner-portal/credit-card-fields/credit-card-loading';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
 
-export default function PaymentMethodAdd(): ReactElement {
+function onPaymentSelectComplete( {
+	successCallback,
+	translate,
+	showSuccessMessage,
+}: {
+	successCallback: () => void;
+	translate: ReturnType< typeof useTranslate >;
+	showSuccessMessage: ( message: string | TranslateResult ) => void;
+} ) {
+	showSuccessMessage( translate( 'Your payment method has been added successfully.' ) );
+	successCallback();
+}
+
+function PaymentMethodAdd(): ReactElement {
+	const dispatch = useDispatch();
 	const translate = useTranslate();
+	const storedCards = useSelector( getAllStoredCards );
+	const { isStripeLoading, stripeLoadingError, stripeConfiguration, stripe } = useStripe();
+
+	const stripeMethod = useCreateStoredCreditCardMethod( {
+		isStripeLoading,
+		stripeLoadingError,
+		stripeConfiguration,
+		stripe,
+		activePayButtonText: translate( 'Save payment method' ) as string,
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **22** times:
`translate( 'Saved payment method' )` *ES Score: 8*